cypher/cypher-docs/src/docs/dev/query-tuning.asciidoc

```diff
@@ -122,6 +125,61 @@ In Enterprise Edition, the Cypher query planner selects the runtime, falling bac
   The interpreted runtime supports all queries, and is the only option in Neo4j Community Edition.
 
 
+[[cypher-replanning]]
+=== Cypher replanning
+
+There may be situations where <<execution-plans,Cypher query planning>> can occur at a non-ideal time.
+For example, when a query must be as fast as possible and a valid plan is already in place.
+
+Cypher replanning occurs in the following circumstances:
+
+* When the query is not in the cache.
+This can either be when the server is first started or restarted, if the cache has recently been cleared, or if <<operations-manual#config_dbms.query_cache_size,dbms.query_cache_size>> was exceeded.
+* When the time has past the <<operations-manual#config_cypher.min_replan_interval,cypher.min_replan_interval>> value, and the database statistics have changed more than the <<operations-manual#config_cypher.statistics_divergence_threshold,cypher.statistics_divergence_threshold>> value.
+
+[NOTE]
+Replanning is not performed for all queries at once; it is performed in the same thread as running the query, and can block the query.
+However, replanning one query does not replan any other queries.
+
+There are three different replan options available:
+
+[options="header",cols="m,a,a"]
+|===
+|Option
+|Description
+|Default
+
+|replan=default
+|This is the planning and replanning option as described above.
+|X
+
+|replan=force
+|This will force a replan, even if the plan is valid according to the planning rules.
+Once the new plan is complete, it replaces the existing one in the query cache.
+|
+
+|replan=skip
+|If a valid plan already exists, it will be used even if the planning rules would normally dictate that it should be replanned.
+|
+|===
+
+The replan option is prepended to queries.
+For example:
+
+[source, cypher]
+----
+CYPHER replan=force MATCH ...
+----
+
+In a mixed workload, you can force replanning by using the Cypher `EXPLAIN` commands.
+For example:
+
+[source, cypher]
+----
+CYPHER replan=force EXPLAIN MATCH ...
+----
+
```

Comment:
I would add something in the sense of:

"
This can be useful to schedule replanning of queries which are expensive to plan, at known times of low load. Using `EXPLAIN` will make sure the query is only planned, but not executed. During times of known high load, `replan=skip` is useful to not introduce unwanted latency spikes.
"